specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2017-10-01/containerregistry.json

```diff
@@ -1014,6 +1055,60 @@
     }
   },
   "definitions": {
+    "ImportImageParameters": {
+      "required": [
```

Comment:
Please add `"type": "object"`.

Reply:
@olydis We were instructed from the very first swagger `2016-06-27-preview` version that this `"type": "object"` should be removed from here since it's redundant. We don't have it in all the versions since then, and looking at other swaggers such as [Storage](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/Microsoft.Storage/stable/2017-10-01/storage.json#L1365), [Compute](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2017-12-01/compute.json#L5651), etc. they don't have this either. Is this a new requirement? Or do we need to update other places if `"type": "object"` is now required?

Reply:
Yeeeeeah so somehow the idea that `type: object` is redundant was a false assumption we made for a long long time. Swagger follows JSON schema, and JSON says that something that doesn't have a `type` has *any* type (if you have stuff like `properties`, it is ignored *unless* you are looking at an `object` but, say, a number also matches such schema! Oops). Long story short, except for a handful of cases across all of Azure, stuff really has a `type`! No worries about the past Swaggers and also no worries about the future - our ancient code generation was created with the same false assumption so it behaves perfectly fine regardless of `type` or not. But external tools may take it more seriously, and ultimately our job is to provide valid Swaggers to our customers... and currently, if a PUT's body accepts a `{ "properties": { ... } }` then that means customers are allowed to send a `42` over the wire. We have a work item on our side to normalize the entire repo to have `type: object` where it belongs, but whenever I see it in a PR, I ask for it... but I don't expect you to spend more efforts on that since indeed we gave the wrong guidance for years!

Reply:
We use a global filter in swashbuckle to get rid of all the `"type": "object"` so I guess it's actually easier for us to consistently have it in this swagger, rather than having a special case for the new API/definitions, but we probably don't want to touch old swaggers as you also mentioned.

Reply:
ah sure, that makes perfect sense 👍